### PR TITLE
[JavaScript] Fix check in getStringValue

### LINKED ISSUE #3237: Number Input default value undefined when value is set to zero
# Platform

* JavaScript

# Author or host
Host Rendering Cards

# Version of SDK

npm Adaptive Cards v1.2.0

# Issue

Per issue https://github.com/microsoft/BotFramework-WebChat/issues/2192

Adaptive inputs of type `Input.Number` don't display the value `0`.

image

## To Reproduce

Steps to reproduce:

1. Go to [Adaptive Cards Designer](https://adaptivecards.io/designer/)
2. Select `Bot Framework WebChat` as the host app
3. Paste the following json in the `Card Payload Editor`:
~~~
{
    "type": "AdaptiveCard",
    "actions": [
      {
        "title": "Submit",
        "type": "Action.Submit"
      }
    ],
    "body": [
      {
        "text": "Input 1",
        "type": "TextBlock"
      },
      {
        "id": "InputOne",
        "placeholder": "Input 1 Placeholder",
        "type": "Input.Number",
        "value": 10
      },
      {
        "text": "Input 2",
        "type": "TextBlock"
      },
      {
        "id": "InputTwo",
        "placeholder": "Input 2 Placeholder",
        "type": "Input.Number",
        "value": -10
      },
      {
        "text": "Input 3",
        "type": "TextBlock"
      },
      {
        "id": "InputThree",
        "placeholder": "Input 3 Placeholder",
        "type": "Input.Number",
        "value": 0
      }
    ],
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.0"
}
~~~
4. See that the third input is not displaying the expected value 

# Likely cause
It looks like Adaptive Card's `Input` class's `parse` method uses the `getStringValue` utility to set the input's default value. The function uses a ternary to either return the value set in the Adaptive Card JSON as a string or the default value which would be `undefined`. In the case of the value being `0`, the `getStringValue` function would return `undefined`.

**[Input Parse Method](https://github.com/microsoft/AdaptiveCards/blob/5c1114871186618bb3a2195a85e01b04c55788f0/source/nodejs/adaptivecards/src/card-elements.ts#L2902)**
```javascript
parse(json: any, errors?: Array<HostConfig.IValidationError>) {
    super.parse(json, errors);

    this.id = Utils.getStringValue(json["id"]);
    this.defaultValue = Utils.getStringValue(json["value"]);

    if (AdaptiveCard.useBuiltInInputValidation) {
        let jsonValidation = json["validation"];

        if (jsonValidation) {
            this.validation.parse(jsonValidation);
        }
    }
}
```

**[Get String Value Method](https://github.com/microsoft/AdaptiveCards/blob/5c1114871186618bb3a2195a85e01b04c55788f0/source/nodejs/adaptivecards/src/utils.ts#L11)**
```javascript
export function getStringValue(obj: any, defaultValue: string = undefined): string {
    return obj ? obj.toString() : defaultValue;
}
```

# Possible Fix
The `getStringValue` method would likely have to be changed to check if the value is not undefined or equal to zero.

```javascript
export function getStringValue(obj: any, defaultValue: string = undefined): string {
    return (typeof obj !== 'undefined') ? obj.toString() : defaultValue;
}
```

### PR DESCRIPTION
**Pull request by RebeccaAnne**
The check in getStringValue was returning the default (undefined) value for zero as well as for unset. Updated the check to check against undefined so that it won't trigger for zero values.

Fixes #3237

How Verified: Confirmed in designer that card from #3237 now renders as expected.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3350)

### CODE REVIEW

File: source/nodejs/package-lock.json [hunk 1]
```diff
@@ -12759,7 +12759,7 @@
     },
     "tty-browserify": {
       "version": "0.0.0",
-      "resolved": "https://registry.npmjs.org/tty-browserify/-/tty-browserify-0.0.0.tgz",
+      "resolved": "http://registry.npmjs.org/tty-browserify/-/tty-browserify-0.0.0.tgz",
```
Comment (shalinijoshi19) [inline source/nodejs/package-lock.json]: Nit: that's a funny change it's making to the package-lock file (moving away from https to http)  -- wondering if that's an environment setting of some sort or this npm bug ? Quick hunting around resulted in this [https://discuss.circleci.com/t/npm-install-downgrades-package-lock-json-links-from-https-to-http/22508/7](https://discuss.circleci.com/t/npm-install-downgrades-package-lock-json-links-from-https-to-http/22508/7) ..what's your npm version say ? (npm --version) ?

**Comment by shalinijoshi19** [comment]
Nit nit: Can you organize the verification steps into a separate "How Verified" sub-head for consistency? I'll work on getting us a check-in description template going

File: source/nodejs/package-lock.json (hunk 1 above)
Comment (RebeccaAnne) [reply source/nodejs/package-lock.json]: 6.9.0

---
In reply to: [315413547](https://github.com/microsoft/AdaptiveCards/pull/3350#discussion_r315413547) [](ancestors = 315413547)

Comment (shalinijoshi19) [reply source/nodejs/package-lock.json]: I think that should be good -- try this per that last thread -- see if that will get rid of the http changes to package-lock
rm -rf ./node_modules
npm cache clean --force
npm install --prefer-online

---
In reply to: [315415294](https://github.com/microsoft/AdaptiveCards/pull/3350#discussion_r315415294) [](ancestors = 315415294,315413547)

Comment (RebeccaAnne) [reply source/nodejs/package-lock.json]: Tried that, reverted the package-lock file, and rebuilt everything. I'm not getting any changes to the package-lock file anymore so looks like that fixed me. Thanks.

---
In reply to: [315416611](https://github.com/microsoft/AdaptiveCards/pull/3350#discussion_r315416611) [](ancestors = 315416611,315415294,315413547)

**Pull request merged**